flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/SnapshotObserver.java

```diff
@@ -83,6 +84,11 @@ public void observeSavepointStatus(FlinkResourceContext<CR> ctx) {
         var jobStatus = resource.getStatus().getJobStatus();
         var jobId = jobStatus.getJobId();
 
+        if (isBatchJob(ctx, jobId)) {
+            LOG.debug("Skipping checkpoint observation for BATCH job");
+            return;
```

Comment:
It would be good to avoid this extra rest api call all the time and also if in the future batch jobs can also take a checkpoint (which is where Flink is going) this is going to have to be rewritten.

Could we instead catch the checkpoint error in the `observeLatestCheckpoint` method and do this check then?

Also please add a test for this new logic in the observer